joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/AstForTypesCreator.scala

```diff
@@ -248,7 +248,10 @@ trait AstForTypesCreator {
               case d =>
                 astForDeclarator(declaration, d, order)
             }
-          case _ if nodeSignature(declaration) == ";"  => Seq.empty
+          case _ if nodeSignature(declaration) == ";" =>
+            Seq.empty // dangling decls from unresolved macros; we ignore them
```

Comment:
i'd add a test for these

Reply:
I was not yet able to reproduce this in a clean unit test case.
But it did happen e.g., when scanning VLC (wild combination of missing defines and system header files).
We want some actual tests for that - I agree. But I need to investigate this further and do a separate PR.